test/integration/MarketplaceFlow.test.ts

```diff
@@ -0,0 +1,404 @@
+import {
+  AmountsOutMaxFee,
+  approve,
+  Aquarius,
+  balance,
+  Config,
+  Datatoken,
+  downloadFile,
+  Erc20CreateParams,
+  getHash,
+  Nft,
+  NftCreateData,
+  NftFactory,
+  Pool,
+  PoolCreationParams,
+  ProviderFees,
+  ProviderInstance,
+  TokenInOutMarket,
+  ZERO_ADDRESS
+} from '../../src'
+import { getTestConfig, web3 } from '../config'
+import { Addresses, deployContracts } from '../TestContractHandler'
+/// ```
+
+/// <!--
+describe('Marketplace flow tests', async () => {
+  /// -->
+
+  /// Variables and constants needed for the test:
+
+  /// ```Typescript
+  let config: Config
+  let aquarius: Aquarius
+  let providerUrl: any
+  let publisherAccount: string
+  let consumerAccount: string
+  let contracts: Addresses
+  let erc721Address: string
+  let datatokenAddress: string
+  let poolAddress: string
+
+  const NFT_SYMBOL = 'DT1'
+  /// ```
+
+  ///  We will need a file to publish, so here we define the file that we intend to publish.
+  /// ```Typescript
+  const ASSET_URL = [
+    {
+      type: 'url',
+      url: 'https://raw.githubusercontent.com/oceanprotocol/testdatasets/main/shs_dataset_test.txt',
+      method: 'GET'
+    }
+  ]
+  /// ```
+
+  /// Next, we define the metadata that will describe our data asset. This is what we call the DDO
+  /// ```Typescript
+  const DDO = {
+    '@context': ['https://w3id.org/did/v1'],
+    id: '',
+    version: '4.0.0',
+    chainId: 4,
+    nftAddress: '0x0',
+    metadata: {
+      created: '2021-12-20T14:35:20Z',
+      updated: '2021-12-20T14:35:20Z',
+      type: 'dataset',
+      name: 'dataset-name',
+      description: 'Ocean protocol test dataset description',
+      author: 'oceanprotocol-team',
+      license: 'MIT'
+    },
+    services: [
+      {
+        id: 'testFakeId',
+        type: 'access',
+        files: '',
+        datatokenAddress: '0x0',
+        serviceEndpoint: 'https://providerv4.rinkeby.oceanprotocol.com',
+        timeout: 0
+      }
+    ]
+  }
+  /// ```
+
+  /// We load the configuration:
+  /// ```Typescript
+  before(async () => {
+    config = await getTestConfig(web3)
+    aquarius = new Aquarius(config.metadataCacheUri)
+    providerUrl = config.providerUri
+
+    console.log(`Aquarius URL: ${config.metadataCacheUri}`)
+    console.log(`Provider URL: ${providerUrl}`)
+  })
+  /// ```
+
+  it('Initialize accounts', async () => {
+    /// ```Typescript
+    const accounts = await web3.eth.getAccounts()
+    publisherAccount = accounts[0]
+    consumerAccount = accounts[1]
+
+    console.log(`Publisher account address: ${publisherAccount}`)
+    console.log(`Consumer account address: ${consumerAccount}`)
+  })
+  /// ```
+
+  it('Deploy contracts', async () => {
+    /// ```Typescript
+    contracts = await deployContracts(web3, publisherAccount)
+  })
+  /// ```
+
+  it('Publish a dataset (create NFT + ERC20) with a liquidity pool', async () => {
```

Comment:
I would expand this so that you create two more assets. One with fixed pricing and one with free pricing